lib/loaders/template-loader.js

```diff
@@ -22,20 +22,38 @@ module.exports = class TemplateLoader extends Loader {
    * @param htmlInfo - The information about the HTML
    */
   load(htmlInfo) {
+    let content;
+
+    // Try to read the file.
     try {
-      const content = frontMatter(fs.readFileSync(htmlInfo.file, 'utf8')); // For some reason the async version take a super long time.
-      this.cache[htmlInfo.file] = content;
-      if(content.attributes.layout){
-        //if this template has a layout, also cache that layout
-        const layoutInfo = HTMLInfo.getLayoutInfo(htmlInfo, content.attributes.layout);
-        this.load(layoutInfo);
-      }
-    } catch
-      (err) {
+      content = fs.readFileSync(htmlInfo.file, 'utf8');
+    } catch (err) {
       const error = new Error(`Reading the file "${htmlInfo.file}" failed.`);
       error.type = 'IO_READING_FAILED';
+      // Pass the filename along so the validator can show the correct file if this error was caused by a layout.
+      error.fileName = htmlInfo.file;
+      throw error;
+    }
+
+    let matter;
+
+    // Try to retrieve frontmatter from the file.
+    try {
+      matter = frontMatter(content); // For some reason the async version take a super long time.
```

Comment:
I think that comment belong a bit higher, no?

Reply:
There is one try for reading the file and one try for getting the frontmatter from the content of the file. I'v made it slightly clearer.